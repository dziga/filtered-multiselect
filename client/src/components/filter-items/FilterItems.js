import React, { Component } from 'react';
import FilterItem from '../filter-item/FilterItem'
import { filter, sort } from './FilterItemsUtils'
import { saveSelected } from '../../services/apiService'
import './FilterItems.scss'

class FilterItems extends Component {
  constructor(props) {
    super(props)
    this.state = {items: props.items}
    this.onItemSelected = this.onItemSelected.bind(this)
  }

  componentWillReceiveProps(props) {
    this.setState({items: props.items})
  }

  render() {
    

    return (
      <div className="items">
        {this.getFilteredAndSortedItems()}
      </div>
    )
  }

  getFilteredAndSortedItems() {
    const {items} = this.state
    const {search} = this.props
    return sort(filter(items, search)).map((item, key) => {
      return <FilterItem key={key} item={item} onItemSelected={this.onItemSelected} />
    })
  }

  onItemSelected(item) {
    item.checked = !item.checked
    this.setState({items: sort(this.state.items)})
    saveSelected(item)
  }
}

export default FilterItems;
