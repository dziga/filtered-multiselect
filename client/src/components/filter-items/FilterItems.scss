.items {
  height: 100%;
  overflow: scroll;
  overflow-x: hidden;
}

::-webkit-scrollbar {
  width: 6px;
}

::-webkit-scrollbar-track {
  box-shadow: inset 0 0 5px grey; 
  border-radius: 10px;
}

::-webkit-scrollbar-thumb {
  background: darkgrey; 
  height: 50px;
  border-radius: 10px;
}