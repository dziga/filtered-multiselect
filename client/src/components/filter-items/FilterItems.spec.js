import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import FilterItems from './FilterItems';

test('filter items should render with given props', () => {
  const items = [{id:1, checked:true, label:'book'}]
  const component = renderer.create(
    <FilterItems items={items} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('click on item should change checked and put it on top', () => {
  const items = [
    {id:1, checked:false, label:'book1'},
    {id:2, checked:false, label:'book2'},
    {id:3, checked:false, label:'book3'}
  ]
  const component = mount((<FilterItems items={items} />))

  component.find('input').at(1).simulate('change', { target: { checked: true } })

  const checkedElement = component.find('input').at(0).props();

  expect(checkedElement.checked).toBeTruthy()
  expect(checkedElement.id).toEqual(2)
})
