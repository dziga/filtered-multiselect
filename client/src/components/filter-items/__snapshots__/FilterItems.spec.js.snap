// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter items should render with given props 1`] = `
<div
  className="items"
>
  <div
    className="item"
  >
    <label>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "book",
          }
        }
      />
      <input
        checked={true}
        id={1}
        name="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </label>
  </div>
</div>
`;
